### DISCUSSION TITLE
Setting default roles per project and adding  new users to one or more of them

### DISCUSSION THREAD

**Discussion by idc77** [zitadel/zitadel#8799] [Q&A] [answered]
I understand that one organization respresents 1 "realm" or "user base".
In this organization I create a project, all projects in that org share the same user base.
I have defined the user and admin roles for 1 project.
I would like to automatically assign every new user of this certain project to the user group.
Banning a user would entail this role to be revoked.

Is it possible to automatically add new users to the user role?
And
Did I create the roles correctly?
- key=user, display name=User, group=User 
- key=admin, display name=Admin, group=Admin 

In short, I would like to set default role memberships for new users, possibly also existing users

**Comment by hifabienne** [comment]
You can use our ZITADEL actions to create such workflows.
So the first qustion is how are the users created.
If they are created through the API, you can use the ZITADEL v2 action, add it on the user create api request and automatically add the role to the user

**Reply by idc77** [reply]
They're not created via API, but like regular users sign up, through the UI I guess.
Are there examples?

**Reply by hifabienne** [reply, marked as answer]
If they register themself, you can use the action v1, we have an example here: https://zitadel.com/docs/apis/actions/code-examples#add-grants-to-users

**Reply by idc77** [reply]
tysm :)

**Reply by JordanPBGN** [reply]
Using actions v2 would the correct approach be to define a `restWebhook` target as per https://zitadel.com/docs/guides/integrate/actions/testing-request and then call the [CreateAuthorization](https://zitadel.com/docs/apis/resources/authorization_service_v2/zitadel-authorization-v-2-authorization-service-create-authorization) endpoint on the v2 AuthorizationService for the corresponding User?

Using Go `user.CreateUserRequest` (from "github.com/zitadel/zitadel-go/v3/pkg/client/zitadel/user/v2") I can't see a way to add grants to the API call using `restCall` like you would for actions v1?
